### PR TITLE
Protect: Update Threat Icons

### PR DESCRIPTION
**Pull request by nateweller**
## Proposed changes:

* Home: Use a red error state in the stat card when there are threats detected.
* Scan: Use the fill variants with color to better emphasize the good/bad results.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Review the icons in Protect.

## Screenshots

Screenshot 2024-12-15 at 12 51 46 PM

Screenshot 2024-12-15 at 12 53 04 PM

Screenshot 2024-12-15 at 12 55 23 PM

### CODE REVIEW

File: projects/plugins/protect/src/js/routes/scan/index.jsx [hunk 1]
```diff
@@ -39,7 +39,11 @@ const ScanPage = () => {
 	}
 
 	const hasHistory = history && history.threats.length;
+<<<<<<< HEAD
 	const showResults = !! status.threats.length || hasHistory;
+=======
+	const showResults = status.threats.length || hasHistory;
+>>>>>>> 1850b0c9a6 (Protect: only show Threats DataViews when scan is in progress or has results to show)
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/routes/scan/index.jsx]: Maybe:
```
const hasActiveThreats = status && status.threats.length;
const hasHistory = history && history.threats.length;
const showResults = hasActiveThreats || hasHistory;
```

**Review by dkmyta — CHANGES_REQUESTED** [review]
Nice enhancement!

A conflict still appears to need fixing in the `ScanPage` component.

We now display the same icon in the `HomePage` scan card for errors and threats - what if we took a similar approach as the `ErrorAdminSectionHero` and made the error icon outlined? Just to add some variation between the two:

Screen Shot on 2024-12-16 at 14-29-18

File: projects/plugins/protect/src/js/routes/scan/index.jsx [hunk 2]
```diff
@@ -39,7 +39,11 @@ const ScanPage = () => {
...
```
Comment (nateweller) [inline projects/plugins/protect/src/js/routes/scan/index.jsx]:
```suggestion
	const hasActiveThreats = status && status.threats.length;
	const hasHistory = history && history.threats.length;
	const showResults = hasActiveThreats || hasHistory;
```

File: projects/plugins/protect/src/js/routes/home/home-statcards.jsx [hunk 3]
```diff
@@ -138,13 +138,7 @@ const HomeStatCards = () => {
 		} else if ( scanError ) {
 			scanIcon = <ShieldIcon variant="error" height={ ICON_HEIGHT } />;
```
Comment (nateweller) [inline projects/plugins/protect/src/js/routes/home/home-statcards.jsx]:
```suggestion
			scanIcon = <ShieldIcon variant="error" outline height={ ICON_HEIGHT } />;
```

File: projects/plugins/protect/src/js/routes/scan/index.jsx (hunk 1 above)
Comment (nateweller) [reply projects/plugins/protect/src/js/routes/scan/index.jsx]: Thanks for catching this! Fixed with https://github.com/Automattic/jetpack/pull/40621#discussion_r1889373979

**Comment by nateweller** [comment]
> We now display the same icon in the `HomePage` scan card for errors and threats - what if we took a similar approach as the `ErrorAdminSectionHero` and made the error icon outlined? Just to add some variation between the two:

Added outline variant here: https://github.com/Automattic/jetpack/pull/40621#discussion_r1889375217

**Review by dkmyta — APPROVED** [review]

**Pull request merged**